File: 21-merge-two-sorted-lists/merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        vector<int> arr;
        ListNode *temp1=list1, *temp2=list2;
        while(temp1!=nullptr){
            arr.push_back(temp1->val);
            temp1=temp1->next;
        }
        while(temp2!=nullptr){
            arr.push_back(temp2->val);
            temp2=temp2->next;
        }
        sort(arr.begin(),arr.end());
        ListNode *newHead = new ListNode();
        ListNode *temp = newHead;
        for(int i=0;i<arr.size();i++){
            temp->next = new ListNode(arr[i]);
            temp=temp->next;
        }
        return newHead->next;
    }
};

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode q=new ListNode(-1);
        ListNode p=q;
        while(list1!=null && list2!=null){
            if(list2.val >= list1.val){
                p.next = list1;
                list1 = list1.next;
            }
            else{
                p.next = list2;
                list2 = list2.next;
            }
            p=p.next;
        }
        if(list1 != null) p.next = list1;
        else p.next = list2;
        return q.next;
    }
}